### PR TITLE
Remove conditional requirement on `network.peer.address` and `network.peer.port`

### PR DESCRIPTION
**Pull request by trask**
I missed these in #342.

Based on https://github.com/open-telemetry/semantic-conventions/pull/342#discussion_r1340004552:

> I disagree with not populating. Then you'll need a database of instrumentation library version/name to find if absence means "not implemented" or "same".

we decided to remove the conditional and always populate these attributes when network-layer instrumentation exists.

## Changes

Remove conditional requirement on `network.peer.address` and `network.peer.port`.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* ~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~

### CODE REVIEW

**Review by AlexanderWert — APPROVED** [review]

**Review by trisch-me — APPROVED** [review]

**Review by mateuszrzeszutek — COMMENTED** [review]
Should we do the same thing for the `server.port` in HTTP server spans? Currently the spec says it's conditionally required:

> If `server.address` is set and the port is not default (80 for `http` scheme, `443` for https).

Should we remove the "If `server.address` is set" part?

**Comment by trask** [comment]
> Should we do the same thing for the `server.port` in HTTP server spans? Currently the spec says it's conditionally required:
> 
> > If `server.address` is set and the port is not default (80 for `http` scheme, `443` for https).
> 
> Should we remove the "If `server.address` is set" part?

My understanding is that if we remove that part, then `server.port` becomes required (when non-default) even if you don't set `server.address` (which we dropped from required to recommended in #114)

**Comment by trask** [comment]
oh, maybe you're suggesting then that we (similarly) make `network.peer.port` conditionally required (on HTTP client spans):

> If `network.peer.address` is set and the port is not default (80 for `http` scheme, `443` for https).

that seems to make sense to me

**Comment by trask** [comment]
> This change makes it possible to set `network.peer.port` without `network.peer.address` -- or, at the very least, that's how I read it. I thought it introduces some asymmetry between how we treat the `server.*` and `network.(peer|local).*` attributes.

ya, it's a good point

do you think of adding similar condition on `network.peer.port` addresses the issue?

> If `network.peer.address` is set and the port is not default (80 for `http` scheme, `443` for https).

**Comment by mateuszrzeszutek** [comment]
> If `network.peer.address` is set

This PR removed that part 😄  -- yes, I think reverting it back would resolve the confusion.
About the port condition: I think it's not relevant to my concern, personally I'd probably prefer to leave it as it is -- while the `server.*` attributes are semantically tied to HTTP (in the context of this document), the network level attributes may provide a bit broader context without HTTP-specific restrictions (e.g. like `network.transport` and `.type` if they weren't opt in).

**Comment by trask** [comment]
> > If `network.peer.address` is set
> 
> This PR removed that part 😄 -- yes, I think reverting it back would resolve the confusion. About the port condition: I think it's not relevant to my concern, personally I'd probably prefer to leave it as it is -- while the `server.*` attributes are semantically tied to HTTP (in the context of this document), the network level attributes may provide a bit broader context without HTTP-specific restrictions (e.g. like `network.transport` and `.type` if they weren't opt in).

👍 fixed

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by lmolkova — APPROVED** [review]

**Comment by trask** [comment]
@AlexanderWert, @lmolkova, @mateuszrzeszutek and @trisch-me - I reset all the approvals because I've made additional changes, ptal, thx

File: model/trace/http.yaml [hunk 1]
```diff
@@ -19,6 +19,10 @@ groups:
       - ref: http.request.method
         sampling_relevant: true
         requirement_level: required
+      - ref: network.peer.address
+      - ref: network.peer.port
+        requirement_level:
+          recommended: If `network.peer.address` is set.
```
Comment (lmolkova) [inline model/trace/http.yaml:25]: wonder if we can move this to the `network.peer.port` definition and removing from everywhere else

**Review by lmolkova — APPROVED** [review]

File: model/trace/http.yaml (hunk 1 above)
Comment (trask) [reply model/trace/http.yaml:25]: it looks like it works (tooling-wise). is it ok to apply `requirement_level` in the registry tho?

Comment (AlexanderWert) [reply model/trace/http.yaml:25]: Let's keep the registry free from `requirement_level` definitions, please.

Requirement level is always context specific, while the intention of the registry is to keep attribute definitions free from context-specific details.

So, I personally like it that we explicitly define the requirement level per specific context / usage of an attribute and would keep it as is.

**Review by AlexanderWert — APPROVED** [review]

**Review by joaopgrassi — APPROVED** [review]

**Pull request merged**